pinot-controller/src/main/java/com/linkedin/pinot/controller/api/restlet/resources/SegmentCompletionUtils.java

```diff
@@ -43,10 +48,32 @@
     return new SegmentCompletionProtocol.Request.Params()
         .withInstanceId(instanceId)
         .withOffset(offset)
-        .withSegmentName(segmentName);
+        .withSegmentName(segmentName)
+        .withSegmentLocation(segmentLocation);
   }
 
-  public static String generateSegmentFileName(String segmentNameStr, long offset, String instanceId) {
-    return segmentNameStr + "##" + offset + "##" +  instanceId;
+  public static String generateSegmentNamePrefix(String segmentName) {
+    return segmentName + ".tmp.";
+  }
+
+  public static String generateSegmentFileName(String segmentNameStr) {
+    return generateSegmentNamePrefix(segmentNameStr) + SegmentCompletionUtils.generateUUID();
```

Comment:
you need to incldue "tmp" in the generated file
<segmentName> + ".tmp" + <uuid>